types/src/transaction/transaction.rs

```diff
@@ -492,6 +493,20 @@ impl HeapSizeOf for Transaction {
     }
 }
 
+fn is_prev_out_duplicated(burns: &[AssetTransferInput], inputs: &[AssetTransferInput]) -> Result<(), Error> {
+    let mut prev_out_set = HashSet::new();
+    for input in inputs.iter().chain(burns) {
+        let prev_out = (input.prev_out.transaction_hash, input.prev_out.index);
+        if !prev_out_set.insert(prev_out) {
+            return Err(Error::DuplicatedPreviousOutput {
+                transaction_hash: input.prev_out.transaction_hash.clone(),
```

Comment:
`H256` derives `Copy`. No need to call `clone`.

Reply:
Fixed.